### ISSUE TITLE
Analysis of PGN130850 (Simrad Event Command)

### ISSUE THREAD

**Issue by duichan** [canboat/canboat#409]
I have been analyzing PGN130850 (Simrad Event Command) using a B&G Vulcan MFD and
believe that the existing CANBoat interpretation needs revising as follows. 

bytes (0,1)	Manufacturer Code | Reserved | Industry Code
byte 2		not known
byte 3		Address (used for AUTOPILOT)
byte 4		Device Group (used for TIMERS and ALARMS)
byte 5		Event Type
bytes (6,7)	Event
byte 8		Parameter1 (event dependent)
bytes (9,10)	Parameter2 (event dependent)
byte 11		Parameter3 (event dependent)
total: 12 bytes

Here are some examples:

AUTOPILOT (Event Type = 0a)
===================
Command		(41,9f), 03, ff, ff, 0a, (1a,00), 02, (d1,06), ff
	                        Address: CAN address of autopilot
	                        Event: (06,00)=standby, (09,00)=heading mode, (0a,00)=GPS mode,
		                          (0f,00)=wind mode, (1a,00)=change heading/wind angle
	                       Parameter1: Direction of change: 02=port, 03=stbd, otherwise ff=undefined
	                       Parameter2: Angle of change [in radians * 10000], otherwise (ff,ff)=undefined
	                       Parameter3: ff=undefined
Note: Autopilot data from [https://github.com/canboat/canboat/issues/272](url)

TIMERS (Event Type = 17)
================
Race Timer	       (41,9f), ff, ff, 01, 17, (3d,00), ff, (ff,ff), ff
	                       Device Group: 00=none, 01=default, 02=Group1, 03=Group2 ... 07=Group6
	                       Event: (3d,00)=start, (3e,00)=stop, (3f,00)=sync, (40,00)=reset
	                       Parameter1, Parameter2, Parameter3: ff=undefined

Trip Timer	               (41,9f), ff, ff, 01, 17, (64,00), 00, (00,00), 00
	                       Device Group: 00=none, 01=default, 02=Group1, 03=Group2 ... 07=Group6
	                       Event: (64,00)=enable, (65,00)=disable, (41,00)=reset all
	                       Parameter1: 00=Trip1, 01=Trip2
	                       Parameter2, Parameter3: 00=unused

SIREN (Event Type = 1f)
===============
Sounding	               (41,9f), ff, ff, ff, 1f, (51,00), c4, (49,29), 0f
	                       Event: (51,00)=siren
	                       Parameter1: c4
	                       Parameter2: (49,29)
	                       Parameter3: 0f
Note: This message is repeated every 0.5 seconds. The values of Parameter1, Parameter2 and Parameter3 do not change; their interpretation is not known.

Muted		       (41,9f), ff, ff, ff, 1f, (51,00), 00, (00,00), 00
	                       Event: (51,00)=siren
	                       Parameter1: 00
	                       Parameter2: (00,00)
	                       Parameter3: 00

AIS VESSELS (Event Type = 24)
===================
Selected	               (41,9f), ff, ff, ff, 24, (6f,00), ff, (ff,ff), ff
	                       Event: (6f,00)=vessel selected
	                       Parameter1, Parameter2, Parameter3: ff=undefined
Note: This message is transmitted whenever a vessel is selected on the chart or in the list of vessels; there is no indication of the specific vessel's identity.

UNDEFINED (Event Type = ff)
===========================
MOB activated	       (41,9f), ff, ff, ff, ff, (6b,00), ff, (ff,ff), ff
	                       Event: (6b,00)=MOB activated
	                       Parameter1, Parameter2, Parameter3: ff=undefined

ALARMS		       (41,9f), ff, ff, 01, ff, (39,00), 12, (00,01), 00
	                       Device Group: 00=none, 01=default, 02=Group1, 03=Group2 ... 07=Group6
	                       Event: (39,00)=raised, (38,00)=acknowledged, (3a,00)=cleared
	                       Parameter1: Source: 0a=shallow water, 0b=deep water, 0c=anchor depth,
                                                              0d=true wind shift, 0e=true wind low, 0f=true wind high,
                                                              12=low voltage, 29=no GPS fix, 71=low boat speed
	                       Parameter2: Severity: (00,00)=red, (00,01)=orange, (00,02)=yellow, (00,03)=blue, (00,04)=green
	                       Parameter3: 00=unused
Note: Other alarms are internal to the MFD with no associated message transmitted - these include:
anchor drift, high voltage, rudder data missing, depth data missing, wind data missing, heading data missing,
dangerous vessel, AIS data lost, waypoint radius, arrival, XTE and all engine status alarms.

**Comment by KristianJerpetjon** [comment]
the 03 and 0B are simnet requests https://canboat.github.io/canboat/canboat.html .. so the AC12 should not be sending the 65305 apart from when replying to requests for mode and status

**Comment by keesverruijt** [comment]
Thanks for the detailed analysis. Status against current `master`, so a future contributor (or anyone with a B&G MFD log) knows exactly what's left:

### Already covered — Autopilot (Event Type `0x0a`)
PGN 130850 was reworked from live AC-42 / NAC-3 captures and the autopilot family is fully modeled, matching your analysis and going a bit further. canboat keys on **byte 4 = "Proprietary ID"** (`SIMNET_EVENT_COMMAND`: 1=Alarm, 2=AP command, 255=Autopilot) and **byte 5 = "Command Type"** (`0x0a`). Your AP events all map: standby `06`, heading `09`, nav/GPS `0a`→10, wind `0f`→15, change-course `1a`→26, Direction (port `02`/stbd `03`), Angle — plus Tack and several Turn patterns.

### Still missing
- **Timers** (Event Type `0x17`) — Race Timer (start `3d`/stop `3e`/sync `3f`/reset `40`) and Trip Timer (enable `64`/disable `65`/reset `41`, Trip1/Trip2). No 130850 variant.
- **Siren** (`0x1f`, Event `0x51`) — not modeled.
- **AIS vessel selected** (`0x24`, Event `0x6f`) — not modeled.
- **MOB activated** (Event `0x6b`) — not modeled.
- **Alarms** — partial/divergent: there is a "Simnet: Alarm" variant, but it uses a different layout and `SIMNET_ALARM` has only 3 unrelated entries. Your richer breakdown — Event raised `39`/acknowledged `38`/cleared `3a`, Source enum (shallow water `0a` … low voltage `12` …), Severity enum (red/orange/yellow/blue/green) — is not captured.

### Why these can't just be dropped in
Your analysis (B&G Vulcan MFD) reads the layout as **byte 4 = Network Group, byte 5 = Event Type** (the family discriminator). canboat currently keys on **byte 4**. They coincide for autopilot (byte 4 = `0xff`), but for the MFD families your discriminator is byte 5 with byte 4 = network group `00–07`. In fact canboat would currently match your timer/alarm examples (byte 4 = `01`) as "Simnet: Alarm" (Proprietary ID = 1) and mislabel them.

A plausible unified model is: byte 4 = `0xff` → autopilot (today's model), byte 4 = `00–07` → MFD event families keyed by byte 5 Event Type. But adding those variants risks colliding with the autopilot model that was validated against real captures, so it needs **B&G MFD capture data** to confirm before implementing. All current `samples/` for 130850 are autopilot-only (`nac3-*`, `ac42-*`, `tp32`); there are no timer/siren/AIS captures.

**What would unblock this:** a raw canboat-format log from a B&G MFD that exercises the race timer, trip timer, siren, AIS-vessel-selected, MOB, and a few alarms. With that, the Timers/Siren/AIS/MOB/structured-Alarm variants can be added and covered by a test. Leaving open until then.

**Comment by keesverruijt** [comment]
Implemented by #695. PGN 130850 is now modeled as the Simnet command channel (byte 5 = SIMNET_EVENT_TYPE discriminator, byte 6 = per-family command), covering the alarm lifecycle (Activate/Acknowledge/Silence/Deactivate/History), race & trip timer events, and MOB Activated/Cancelled (paired with 127233 for position). Also adds 130825, 130845, 130856 and the 182-entry SIMNET_ALARM_ID dictionary. Thanks to everyone who contributed the bus analysis in this thread.
